Objects/listobject.c

```diff
@@ -245,7 +252,7 @@ PyList_GetItem(PyObject *op, Py_ssize_t i)
         PyErr_BadInternalCall();
         return NULL;
     }
-    if (!valid_index(i, Py_SIZE(op))) {
+    if (!valid_index(i, PyList_GET_SIZE(op))) {
```

Comment:
This seems fine, but in general I'd prefer to hold off on changes to the functions that might avoid locking (i.e., won't use critical sections). We don't have all the necessary pieces in place and it's hard to tell if incremental changes actually move the implementation closer to the desired thread-safe version.